src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/MtAmazonDynamoDbBySharedTable.java

```diff
@@ -813,16 +792,22 @@ private DeleteTableResult deleteTableInternal(DeleteTableRequest deleteTableRequ
 
     private void truncateTable(String tableName) {
         if (truncateOnDeleteTable) {
-            ScanResult scanResult = scan(new ScanRequest().withTableName(tableName));
-            log.warn("truncating " + scanResult.getItems().size() + " items from table=" + tableName);
-            for (Map<String, AttributeValue> item : scanResult.getItems()) {
-                deleteItem(new DeleteItemRequest().withTableName(tableName).withKey(getKeyFromItem(item, tableName)));
+            long deletedCount = 0;
+            ScanRequest scanRequest = new ScanRequest().withTableName(tableName);
+            while (true) {
+                ScanResult scanResult = scan(scanRequest);
+                log.warn("truncating " + scanResult.getItems().size() + " items from table=" + tableName);
+                for (Map<String, AttributeValue> item : scanResult.getItems()) {
+                    deleteItem(new DeleteItemRequest().withTableName(tableName)
+                        .withKey(getKeyFromItem(item, tableName)));
+                }
+                deletedCount += scanResult.getItems().size();
+                if (scanResult.getLastEvaluatedKey() == null) {
+                    break;
+                }
+                scanRequest.withExclusiveStartKey(scanResult.getLastEvaluatedKey());
             }
-            log.warn("truncation of " + scanResult.getItems().size() + " items from table=" + tableName
-                + (scanResult.getLastEvaluatedKey() == null
-                ? " complete. "
-                : "but data may have been dropped to the floor, beware."));
-
+            log.warn("truncation of " + deletedCount + " items from table=" + tableName + " complete.");
```

Comment:
Truncating all records of a a deleted table instead of only the first page of a scan